src/librustc/hir/map/mod.rs

```diff
@@ -514,18 +514,21 @@ impl<'hir> Map<'hir> {
         &self.forest.krate.attrs
     }
 
-    pub fn get_module(&self, module: DefId) -> (&'hir Mod, Span, HirId)
-    {
+    pub fn get_module(&self, module: DefId) -> (&'hir Mod, Span, HirId) {
+        self.get_if_module(module).expect("not a module")
+    }
+
+    pub fn get_if_module(&self, module: DefId) -> Option<(&'hir Mod, Span, HirId)> {
```

Comment:
```suggestion
    pub fn opt_get_module(&self, module: DefId) -> Option<(&'hir Mod, Span, HirId)> {
```

Reply:
Actually, from looking at what `update_macro_reachable_mod` does, it may be better to keep this function alone and directly match on the result of `find_entry`/`get_if_local` there.